### DISCUSSION TITLE
Reuse a GitHub Classroom assignment without copying the repo?

### DISCUSSION THREAD

**Discussion by pcantrell** [community/community#31854] [GitHub Education]
I am delighted to see that Classroom has finally added the ability to reuse an assignment! We’ve been dying for this for a long time.

Alas, the feature is useless to us in my department because of one limitation: we do **not** want to copy assignment template repos, only settings.

Why? We organize our GitHub orgs like this:

- `myclass` ← single authoritative source of assignment templates
- `myclass-fall2022` ← student submissions for a specific semester
- `myclass-spring2022`
- etc.

Reasons for this:

- We want a **single source** for the best, most up-to-date version of each assignment. We are constantly tweaking assignments and fixing problems, and we don’t want that hard work to get lost. That’s exactly what GitHub is for, isn’t it?!
- We don’t that org with assignment templates to be littered with student submissions.

Unfortunately, with the new feature, if I attempt to reuse Homework X from Spring 2022, which uses `myclass/hw-x` as its template, the  new feature will create a _whole new template repository_, `myclass-fall2022/hw-x`. Oops! Now I have two copies of the homework template running around!

If wanted to fork the template repo, I would ask to fork the template repo.

My proposal: the “reuse assignment” feature should copy the template repo **only if the template comes from the same org as the one the assignment is being copied from.** When reusing an assignment that points to a template **outside the classroom org**, the new assignment should continue to point at the same template.

**Comment by J569-del** [comment]
Yo, tengo una Minusvalía Total y absoluta, no tengo ningún título académico para poder dar clases en ningún sitio. Ni tengo Empresa, ni centro educativo.

**Reply by J569-del** [reply]
Estas capturas de pantalla demuestran lo que ocurre.
